pyEpiabm/pyEpiabm/sweep/intervention_sweep.py

```diff
@@ -72,9 +79,16 @@ def __call__(self, time):
             num_cases = sum(map(lambda cell: cell.number_infectious(),
                             self._population.cells))
             if intervention.is_active(time, num_cases):
-                intervention(time)
                 if self.intervention_active_status[intervention] is False:
+                    # Update with the single intervention
+                    # (with respect to index)
+                    intervention_type = self.intervention_dict.inverse[
```

Comment:
I agree its unconventional but there are ways to get a key from a dict based on the value, especially not dicts are ordered https://stackoverflow.com/questions/8023306/get-key-by-value-in-dictionary

Reply:
replaced the bidict by the way you suggested